CMakeLists.txt

```diff
@@ -401,26 +401,14 @@ endif()
 
 if(MSVC)
   add_compile_options(
-    /wd4018 # signed/unsigned mismatch
+    /wd4100 # unreferenced parameter
+    /wd4101 # unreferenced local variable
     /wd4102 # unreferenced label
-    /wd4267 # conversion from 'size_t' to 'int', possible loss of data
-    /wd4244 # conversion from '__int64' to 'int', possible loss of data
-    /wd4305 # '=' : truncation from 'double' to 'float'
-    /wd4309 # '=' : truncation of constant value
-    /wd4800 # forcing value to bool 'true' or 'false' (performance warning)
-    /wd4996 # The POSIX name for this item is deprecated.
+    /wd4200 # nonstandard extension used: zero-sized array in struct/union
   )
 
-# this flag causes MSVC to correctly report __cplusplus, which prevents a compiler error
-# caused by some versions of libsndfile's C++ header redefining nullptr as NULL.
-# See https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/
-  add_compile_options("/Zc:__cplusplus")
```

Comment:
you probably want to keep this flag: without it __cplusplus is *always* defined to 199711L

https://learn.microsoft.com/en-us/cpp/build/reference/zc-cplusplus?view=msvc-170